src/Connections.Abstractions/Features/IConnectionLifetimeFeature.cs

```diff
@@ -1,13 +1,20 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
 using System.Threading;
 
 namespace Microsoft.AspNetCore.Connections.Features
 {
     public interface IConnectionLifetimeFeature
     {
         CancellationToken ConnectionClosed { get; set; }
+
         void Abort();
+
+        // REVIEW: This is technically a breaking change for implementors of this interface,
+        // but realistically this only affects transport implementations which today must
+        // implement other pubternal interfaces or facades which I doubt exist.
+        void Abort(ConnectionAbortedException abortReason);
```

Comment:
@muratg

Reply:
This seems a bit weird. Interface breaking change aside, why `ConnectionAbortedException`

Reply:
The ConnectionAbortedException is to maintain breaking compatibility. Previously, when the output PipeWriter was completed with an error, it would be swallowed by the libuv transport and turned into a ConnectionAbortedException with the (oh so helpful) "The connection was aborted" message. Now at least the message is more helpful.

Now that we actually rethrow the exception passed to abort to any pending input readers, it's important to call Abort with a ConnectionAbortedException for back compat. When I first made this change some tests failed because they were expecting an OperationCanceledException (which a ConnectionAbortedException is) instead of a TimeoutException.

I changed the type in the method signature so the compiler makes it impossible to break compat unintentionally, but we could change it back to taking any exception and just be careful when calling it.

Reply:
I still think it makes  moresense for pending readers to observe a `ConnectionAbortedException` when the connection is aborted due to a call to `IConnectionLifetimeFeature.Abort()`. It's nice when you're reading a request body to know that every exception will be either an `IOException` or an `OperationCanceledException`. Having to handle completely arbitrary exceptions like `TimeoutException`s isn't nice.

Reply:
I don't have a problem with the raised exception to be `ConnectionAbortedException`, I don't think it needs to be the originating exception type though. You can wrap the exception passed into a ConnectionAbortedException

Reply:
Also, you can set the inner exception to whatever you want. Having a call to Abort() result in a ConnectionAbortedException just seems logical.

Reply:
> You can wrap the exception passed into a ConnectionAbortedException

This is a transport feature, so every transport would have to have the exact same wrapping logic, and there would be no compile time guarantees the transport gets it right. I don't see why we wouldn't just keep the common logic in the core assembly.